src/providers/BackgroundReplacementProvider/index.tsx

```diff
@@ -104,6 +111,18 @@ const BackgroundReplacementProvider: FC<Props> = ({
     };
   }, [replacementSpec, replacementOptions]);
 
+  useEffect(() => {
+    if (processor && observer) {
+      (processor as BackgroundReplacementProcessor).addObserver(observer);
```

Comment:
Would it be better to update the State type itself based on usage? Believe the processor in this provider will always be `BackgroundReplacementProcessor` and it extends `VideoFrameProcessor`.
https://github.com/aws/amazon-chime-sdk-js/blob/main/src/backgroundreplacementprocessor/BackgroundReplacementProcessor.ts#L10

I think we can do that and is better over typecasting. If `VideoFrameProcessor` state type is needed then here instead of casting should we check for `instanceof` to make sure `processor` here is actually an instance of `BackgroundReplacementProcessor`. What say?

Reply:
I tried instanceof, but I guess because BackgroundReplacementProcessor is an interface, it doesnt work.

Reply:
Ohh I see, missed the interface and took it for class, my bad. Well, in that case should we update the type from `VideoFrameProcessor` to `BackgroundReplacementProcessor`? The reason is say if any other type of processor is passed and at runtime we cannot cast it then that might be an error.

Reply:
done

Reply:
Just to note, I dont think this is a breaking change as we are actually expanding the types and not shrinking it so believe we should be good on that.